distributed/scheduler.py

```diff
@@ -7419,6 +7419,12 @@ def log_event(self, name, msg):
         else:
             self.events[name].append(event)
             self.event_counts[name] += 1
+            if name == "print":
+                for comm in self.client_comms.values():
+                    comm.send({"op": "print", "message": msg, **kwargs})
+            if name == "warn":
+                for comm in self.client_comms.values():
+                    comm.send({"op": "warn", "warning": msg, **kwargs})
```

Comment:
In the case that we're logging the same message under multiple topics (e.g. `name=["model_score", "print"]`) we won't send `print` or `warn` ops to the client. I'd suggest modifying the logic above to be something along the lines of 

```python
if not isinstance(name, list):
    name = [name]
for n in name:
    ... # logic in the current `else:` branch here 
```

Reply:
I went the other way, and called `self.log_event(...)` a few times, but I agree that not duplicating the logic twice is a good idea.